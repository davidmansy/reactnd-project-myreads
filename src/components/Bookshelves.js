import React from 'react';
import data from '../data/data.json';
import Bookshelf from './Bookshelf';

function Bookshelves(props) {
  const shelves = data.shelves;
  const { books, handleShelfChange, isLoading } = props;

  return (
    <div className="list-books-content">
      <div>
        {isLoading ? (
          <div>Loading...</div>
        ) : (
          shelves.map(shelf => (
            <Bookshelf
              key={shelf.id}
              shelf={shelf}
              books={books}
              handleShelfChange={handleShelfChange}
            />
          ))
        )}
      </div>
    </div>
  );
}

export default Bookshelves;
