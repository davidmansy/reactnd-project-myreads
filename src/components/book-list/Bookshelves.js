import React from 'react';
import data from '../../data/data.json';
import Bookshelf from './Bookshelf';
import Loading from '../common/Loading/index';

function Bookshelves(props) {
  const shelves = data.shelves;
  const { books, handleShelfChange, isLoading } = props;

  return (
    <div className="list-books-content">
      <div className="bookshelves">
        {isLoading ? (
          <Loading />
        ) : (
          shelves.map(shelf => (
            <Bookshelf
              key={shelf.id}
              shelf={shelf}
              books={books}
              handleShelfChange={handleShelfChange}
            />
          ))
        )}
      </div>
    </div>
  );
}

export default Bookshelves;
