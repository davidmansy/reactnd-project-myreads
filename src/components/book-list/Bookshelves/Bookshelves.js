import React from 'react';
import data from '../../../data/data.json';
import Bookshelf from '../Bookshelf/Bookshelf';
import Loading from '../../common/Loading/index';
import PropTypes from 'prop-types';

function Bookshelves(props) {
  const shelves = data.shelves;
  const { books, handleShelfChange, isLoading } = props;

  return (
    <div className="list-books-content">
      <div className="bookshelves">
        {isLoading ? (
          <Loading />
        ) : (
          shelves.map(shelf => (
            <Bookshelf
              key={shelf.id}
              shelf={shelf}
              books={books}
              handleShelfChange={handleShelfChange}
            />
          ))
        )}
      </div>
    </div>
  );
}

Bookshelves.propTypes = {
  books: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleShelfChange: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired
};

export default Bookshelves;
