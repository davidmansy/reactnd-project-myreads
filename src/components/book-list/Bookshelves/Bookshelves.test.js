import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Bookshelves from './Bookshelves';
import { MemoryRouter } from 'react-router-dom';
import { cleanup } from 'react-testing-library';

afterEach(cleanup);

describe('Bookshelves', () => {
  const props = {
    books: [],
    handleShelfChange: () => {},
    isLoading: false
  };

  it('renders', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Bookshelves {...props} />
      </MemoryRouter>,
      container
    );
  });

  it('snapshots', () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter>
        <Bookshelves {...props} />
      </MemoryRouter>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
