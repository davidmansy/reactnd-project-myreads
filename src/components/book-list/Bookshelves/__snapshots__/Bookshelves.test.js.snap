// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookshelves snapshots 1`] = `
<div
  className="list-books-content"
>
  <div
    className="bookshelves"
  >
    <div
      className="bookshelf"
    >
      <h3
        className="bookshelf-title"
      >
        Want to Read
      </h3>
      <div
        className="bookshelf-books"
      >
        <div
          className="bookshelf-empty"
        >
          There are not books available in this shelf
        </div>
      </div>
    </div>
    <div
      className="bookshelf"
    >
      <h3
        className="bookshelf-title"
      >
        Currently Reading
      </h3>
      <div
        className="bookshelf-books"
      >
        <div
          className="bookshelf-empty"
        >
          There are not books available in this shelf
        </div>
      </div>
    </div>
    <div
      className="bookshelf"
    >
      <h3
        className="bookshelf-title"
      >
        Read
      </h3>
      <div
        className="bookshelf-books"
      >
        <div
          className="bookshelf-empty"
        >
          There are not books available in this shelf
        </div>
      </div>
    </div>
  </div>
</div>
`;
